crates/flux-middle/src/rty/projections.rs

```diff
@@ -44,6 +44,18 @@ impl<'sess, 'tcx, 'cx> Normalizer<'sess, 'tcx, 'cx> {
         Ok(Normalizer { genv, selcx, def_id: callsite_def_id, param_env })
     }
 
+    fn normalize_alias_pred(&mut self, alias_pred: &AliasPred, ty: &Ty) -> QueryResult<Ty> {
+        if let Some(impl_id) = self.impl_id_of_alias_ty(alias_pred)?
+            && let Some(pred) = self.genv.assoc_predicate_of(impl_id, alias_pred.name)?
+            && let AssocPredicateKind::Impl(body) = pred.kind
+        {
+            let expr = body.replace_bound_exprs(&alias_pred.refine_args);
```

Comment:
If this body is represented with a `Binder` the more correct thing to do is to represent `AliasPred` separate from `RefineArgs`. Note also that we are cheating here because `body` is implicitly under an `EarlyBinder` and we are skipping it, but that's ok because generics will only appear in sorts which are being removed because the binder is being removed.

If this comment doesn't make sense. Feel free to ignore, I'll fix it latter.

Reply:
Hmm I don't think I followed, but I did split the `RefineArgs` out of `AliasPred`, by which I presumed you meant (from the HackMD)

```rust
pub enum AliasPred { 
    trait_id: DefId,
    name: Symbol,
    generic_args: GenericArgs,
}

pub enum Pred {
    Expr(Expr),
    Alias(AliasPred, RefineArgs)
}
```